include/circt/Dialect/HW/HWModuleGraph.h

```diff
@@ -0,0 +1,189 @@
+//===- HWModuleGraph.h - HWModule graph -------------------------*- C++ -*-===//
```

Comment:
I am uncomfortable putting this amount of logic in a header file. Since AFAICT these graphtraits are only used in the cpp file below, can the contents of this header file be moved there?

Reply:
We could, but this would then break with convention in that `DOTGraphTrait`s tend to be placed next to their `GraphTrait` counterpart.

Reply:
Why not move the entire contents of this file into a CPP file?